tests/h2-support/src/trace.rs

```diff
@@ -1,41 +1,19 @@
-use std::{io, str};
 pub use tracing;
 pub use tracing_subscriber;
 
-pub fn init() -> tracing::dispatcher::DefaultGuard {
-    tracing::subscriber::set_default(
-        tracing_subscriber::fmt()
-            .with_max_level(tracing::Level::TRACE)
-            .with_span_events(tracing_subscriber::fmt::format::FmtSpan::CLOSE)
-            .with_writer(PrintlnWriter { _p: () })
-            .finish(),
-    )
-}
-
-struct PrintlnWriter {
-    _p: (),
-}
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
 
-impl tracing_subscriber::fmt::MakeWriter for PrintlnWriter {
-    type Writer = PrintlnWriter;
-    fn make_writer(&self) -> Self::Writer {
-        PrintlnWriter { _p: () }
-    }
-}
-
-impl io::Write for PrintlnWriter {
-    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-        let s = str::from_utf8(buf).map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, e))?;
-        println!("{}", s);
-        Ok(s.len())
-    }
-
-    fn write_fmt(&mut self, fmt: std::fmt::Arguments<'_>) -> io::Result<()> {
-        println!("{}", fmt);
-        Ok(())
-    }
+pub fn init() -> tracing::dispatcher::DefaultGuard {
+    let use_colors = atty::is(atty::Stream::Stderr);
+    let layer = tracing_tree::HierarchicalLayer::default()
+        .with_writer(std::io::stderr)
```

Comment:
the previous code used `println!` rather than writing directly to an IO stream so that the log output was captured by libtest's output capturing. I think we might want to change this code to do the same. That way, we can use `cargo test -- --show-output`, and get the output categorized by test, rather than having it all dumped to the console in a big mess when tests run in parallel. Also, it will only print logs for failing tests by default, which should make a `cargo test` run less noisy out of the box.

`tracing-subscriber` has a [`TestWriter` type](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.TestWriter.html) that does this (I believe it was added after the `PrintlnWriter` here was written). 

Since that uses `println!` rather than `eprintln!`, I think we'll want to check if stdout is a tty, rather than stderr:
```suggestion
    let use_colors = atty::is(atty::Stream::Stdout);
    let layer = tracing_tree::HierarchicalLayer::default()
        .with_writer(tracing_subscriber::fmt::writer::TestWriter::default())
```